liberty-maven-plugin/src/main/java/io/openliberty/tools/maven/server/DevMojo.java

```diff
@@ -966,6 +967,27 @@ public void checkConfigFile(File configFile, File serverDir) {
             }
         }
 
+        @Override
+        public boolean serverFeaturesModified(File configDirectory) {
+            ServerFeatureUtil servUtil = getServerFeatureUtil();
+            servUtil.setSuppressLogs(true); // suppress logs from ServerFeatureUtil, otherwise will flood dev console
+            // get server features from the config directory, exclude generated-features.xml
+            Set<String> generatedFiles = new HashSet<String>();
+            generatedFiles.add(BinaryScannerUtil.GENERATED_FEATURES_FILE_NAME);
+            // if serverXmlFile is null, getServerFeatures will use the default server.xml in the configDirectory
+            Set<String> features = servUtil.getServerFeatures(configDirectory, serverXmlFile,
+                    new HashMap<String, File>(), generatedFiles);
+
+            // exclude generated features from the features list
+            Set<String> generatedFeatures = servUtil.getServerXmlFeatures(null,
+                    new File(configDirectory, BinaryScannerUtil.GENERATED_FEATURES_FILE_PATH), null, null);
+            servUtil.setSuppressLogs(false); // re-enable logs from ServerFeatureUtil
+            if (features != null && generatedFeatures != null) {
+                features.removeAll(generatedFeatures);
+            }
+            return servUtil.featuresModified(features, existingFeatures);
```

Comment:
As discussed with team, this is always returning true on any change because of the exclusion of the generated-features.xml features.